scripts/zones/Palborough_Mines/npcs/_3z7.lua

```diff
@@ -6,17 +6,48 @@ require("scripts/globals/status")
 -----------------------------------
 local entity = {}
 
+entity.isBusy = false
+
 entity.onTrade = function(player, npc, trade)
 end
 
 entity.onTrigger = function(player, npc)
-    RunElevator(xi.elevator.PALBOROUGH_MINES_LIFT)
+    local state = player:getElevatorState(xi.elevator.PALBOROUGH_MINES_LIFT)
```

Comment:
If the binding doesn't rely on the player or the npc in this interaction, should it instead be part of lua_zone and then you call through npc:getZone():getElevetorStuff(id)?

Reply:
`RunElevator` lives in luautils so I put it there, how's this?

Reply:
if I were to (hypothetical) redo the entire system from scratch, all of it would live in baseentity. every single part of a elevator is an npc. down to the moving gears, invisible doors, and platform itself - all npc's and their animation states. the sole contribu5tion teh zone makes is to tell them when to start their repeating task for the automatic ones. Pull-lever ones like this rely on a player interaction instead.